gensim/models/tfidfmodel.py

```diff
@@ -78,10 +81,47 @@ def __init__(self, corpus=None, id2word=None, dictionary=None,
         and it will be used to directly construct the inverse document frequency
         mapping (then `corpus`, if specified, is ignored).
         """
-        self.normalize = normalize
         self.id2word = id2word
-        self.wlocal, self.wglobal = wlocal, wglobal
+        self.wlocal, self.wglobal, self.normalize = wlocal, wglobal, normalize
         self.num_docs, self.num_nnz, self.idfs = None, None, None
+        n_tf, n_df, n_n = smartirs
+        self.smartirs = smartirs
+
+        if wlocal is None:
+            def wlocal(tf, mean=None, _max=None):
+                if n_tf == "n":
+                    return tf
+                elif n_tf == "l":
+                    return 1 + math.log(tf)
+                elif n_tf == "a":
+                    return 0.5 + (0.5 * tf / _max)
+                elif n_tf == "b":
+                    return 1 if tf > 0 else 0
+                elif n_tf == "L":
+                    return (1 + math.log(tf)) / (1 + math.log(mean))
```

Comment:
What happens if the value is none of the enumerated options? Will the error make sense to the user?

Reply:
[`resolve_weights`](https://github.com/RaRe-Technologies/gensim/pull/1791/files#diff-896942304f9417b1b8842823b19d7431R19) take care of that.